internal/signer/linux/pkcs11/pkcs11.go

```diff
@@ -139,3 +145,52 @@ func (k *Key) Public() crypto.PublicKey {
 func (k *Key) Sign(_ io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
 	return k.signer.Sign(nil, digest, opts)
 }
+
+func (k *Key) Encrypt(data []byte) ([]byte, error) {
+	publicKey := k.Public()
+	_, ok := publicKey.(*rsa.PublicKey)
+	if ok {
+		return k.encryptRSAGoPKCS11(data)
+	}
+	_, ok = publicKey.(*ecdsa.PublicKey)
+	if ok {
+		// TODO: implement encryption for ec keys
+		return nil, nil
+	}
+	return nil, fmt.Errorf("encrypt error: Unsupported key type")
+}
+
+func (k *Key) Decrypt(encryptedData []byte) ([]byte, error) {
+	publicKey := k.Public()
+	_, ok := publicKey.(*rsa.PublicKey)
+	if ok {
+		return k.decryptRSAGoPKCS11(encryptedData)
+	}
+	_, ok = publicKey.(*ecdsa.PublicKey)
+	if ok {
+		// TODO: implement decryption for ec keys
+		return nil, nil
+	}
+	return nil, fmt.Errorf("decrypt error: Unsupported key type")
+}
+
+func (k *Key) encryptRSA(hash hash.Hash, data []byte) ([]byte, error) {
+	publicKey := k.Public()
+	rsaPubKey := publicKey.(*rsa.PublicKey)
+	return rsa.EncryptOAEP(hash, rand.Reader, rsaPubKey, data, nil)
+}
+
+func (k *Key) encryptRSAGoPKCS11(data []byte) ([]byte, error) {
+	publicKeyFilter := pkcs11.Filter{pkcs11.ClassPublicKey, ""}
+	pubObjs, err := (k.slot).Objects(publicKeyFilter)
+	pubObj := pubObjs[0]
+	if err != nil {
+		return nil, fmt.Errorf("encryptRSAGoPKCS11 public key error, %v", err)
```

Comment:
Use the pattern "What failed. Why did it fail. What does the mean for the consumer"

Reply:
Okay, I updated it to: ("encryptRSAWithPKCS11 error retrieving public key: %v", err)

Reply:
I was thinking something more along the lines of `"Failed to encrypt. Failed to find a public key. Cannot perform encryption without a public key. Downstream failure due to %v."`

Remember, most of your customers will be people that don't know what ECP is, and it will help maintainers and customers understand errors if the logs are detailed.